### ISSUE TITLE
matplotlib 3.3 is now available on pip causing breakage

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#864]
Here's an example of an error

```
 ValueError: 'steps-mid' is not a valid value for ls; supported values are '-', '--', '-.', ':', 'None', ' ', '', 'solid', 'dashed', 'dashdot', 'dotted'
```

**Comment by DougBurke** [comment]
Here's the two failures I get with mpl 3.3 installed (but not astropy; if I install astropy you get 7 more failures, 6 of them like the followng and one a difference in a preference-style test).

Note that with #693 the tests pass.

```
========================================== FAILURES ==========================================
___________________________ test_numpy_histogram_density_vs_normed ___________________________

    def test_numpy_histogram_density_vs_normed():
        from sherpa.astro import ui

        ui.load_arrays(1, [1, 2, 3], [1, 2, 3])
        ui.set_source('const1d.c')
        c = ui.get_model_component('c')
        ui.fit()
        res = ui.eqwidth(c, c+c, error=True)
>       ui.plot_pdf(res[4])

sherpa/plot/tests/test_plot.py:569:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
<string>:1: in plot_pdf
    ???
sherpa/ui/utils.py:13284: in plot_pdf
    self._pdfplot.plot(overplot=overplot, clearwindow=clearwindow,
sherpa/plot/__init__.py:481: in plot
    Histogram.plot(self, self.xlo, self.xhi, self.y, title=self.title,
sherpa/plot/__init__.py:406: in plot
    backend.histo(xlo, xhi, y, yerr=yerr, title=title,
sherpa/plot/pylab_backend.py:154: in histo
    plot(xmid, y, yerr=yerr, xerr=None,
sherpa/plot/pylab_backend.py:322: in plot
    axes.plot(x, y,
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_axes.py:1743: in plot
    lines = [*self._get_lines(*args, data=data, **kwargs)]
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_base.py:273: in __call__
    yield from self._plot_args(this, kwargs)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_base.py:418: in _plot_args
    return [func(x[:, j % ncx], y[:, j % ncy], kw, kwargs)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_base.py:418: in <listcomp>
    return [func(x[:, j % ncx], y[:, j % ncy], kw, kwargs)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_base.py:312: in _makeline
    seg = mlines.Line2D(x, y, **kw)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/lines.py:363: in __init__
    self.set_linestyle(linestyle)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/lines.py:1133: in set_linestyle
    cbook._check_in_list([*self._lineStyles, *ls_mapper_r], ls=ls)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

_values = ['-', '--', '-.', ':', 'None', ' ', ...], kwargs = {'ls': 'steps-mid'}
values = ['-', '--', '-.', ':', 'None', ' ', ...], k = 'ls', v = 'steps-mid'

    def _check_in_list(_values, **kwargs):
        """
        For each *key, value* pair in *kwargs*, check that *value* is in *_values*;
        if not, raise an appropriate ValueError.

        Examples
        --------
        >>> cbook._check_in_list(["foo", "bar"], arg=arg, other_arg=other_arg)
        """
        values = _values
        for k, v in kwargs.items():
            if v not in values:
>               raise ValueError(
                    "{!r} is not a valid value for {}; supported values are {}"
                    .format(v, k, ', '.join(map(repr, values))))
E               ValueError: 'steps-mid' is not a valid value for ls; supported values are '-', '--', '-.', ':', 'None', ' ', '', 'solid', 'dashed', 'dashdot', 'dotted'

../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/cbook/__init__.py:2248: ValueError
__________________________________ test_plot_xdf[plot_pdf] ___________________________________

plotfunc = <function plot_pdf at 0x7f2602171a60>

    @pytest.mark.parametrize("plotfunc", [ui.plot_cdf, ui.plot_pdf])
    def test_plot_xdf(plotfunc):
        """Very basic check we can call plot_cdf/pdf

        This can be run even without a plotting backend available.
        """

        pvals = [0.1, 0.2, 0.1, 0.4, 0.3, 0.2, 0.1, 0.6]
>       plotfunc(pvals)

sherpa/ui/tests/test_ui_plot.py:777:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
<string>:1: in plot_pdf
    ???
sherpa/ui/utils.py:13284: in plot_pdf
    self._pdfplot.plot(overplot=overplot, clearwindow=clearwindow,
sherpa/plot/__init__.py:481: in plot
    Histogram.plot(self, self.xlo, self.xhi, self.y, title=self.title,
sherpa/plot/__init__.py:406: in plot
    backend.histo(xlo, xhi, y, yerr=yerr, title=title,
sherpa/plot/pylab_backend.py:154: in histo
    plot(xmid, y, yerr=yerr, xerr=None,
sherpa/plot/pylab_backend.py:322: in plot
    axes.plot(x, y,
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_axes.py:1743: in plot
    lines = [*self._get_lines(*args, data=data, **kwargs)]
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_base.py:273: in __call__
    yield from self._plot_args(this, kwargs)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_base.py:418: in _plot_args
    return [func(x[:, j % ncx], y[:, j % ncy], kw, kwargs)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_base.py:418: in <listcomp>
    return [func(x[:, j % ncx], y[:, j % ncy], kw, kwargs)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/axes/_base.py:312: in _makeline
    seg = mlines.Line2D(x, y, **kw)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/lines.py:363: in __init__
    self.set_linestyle(linestyle)
../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/lines.py:1133: in set_linestyle
    cbook._check_in_list([*self._lineStyles, *ls_mapper_r], ls=ls)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

_values = ['-', '--', '-.', ':', 'None', ' ', ...], kwargs = {'ls': 'steps-mid'}
values = ['-', '--', '-.', ':', 'None', ' ', ...], k = 'ls', v = 'steps-mid'

    def _check_in_list(_values, **kwargs):
        """
        For each *key, value* pair in *kwargs*, check that *value* is in *_values*;
        if not, raise an appropriate ValueError.

        Examples
        --------
        >>> cbook._check_in_list(["foo", "bar"], arg=arg, other_arg=other_arg)
        """
        values = _values
        for k, v in kwargs.items():
            if v not in values:
>               raise ValueError(
                    "{!r} is not a valid value for {}; supported values are {}"
                    .format(v, k, ', '.join(map(repr, values))))
E               ValueError: 'steps-mid' is not a valid value for ls; supported values are '-', '--', '-.', ':', 'None', ' ', '', 'solid', 'dashed', 'dashdot', 'dotted'

../../anaconda/envs/sherpa-xspec-12101/lib/python3.8/site-packages/matplotlib/cbook/__init__.py:2248: ValueError
```

**Comment by DougBurke** [comment]
Well, all-but-one of the failures go away due to #693 (and I need to run with an XSPEC build as a final check)

**Comment by DougBurke** [comment]
Looks like this has made it's way to recent (python 3.7) conda builds now.

**Comment by DougBurke** [comment]
Closed by #693
